.pre-commit-config.yaml

```diff
@@ -0,0 +1,21 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+exclude: /test/.*$(?<!\.ql)(?<!\.qll)(?<!\.qlref)
+repos:
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v3.2.0
+    hooks:
+    -   id: trailing-whitespace
+    -   id: end-of-file-fixer
+-   repo: local
+    hooks:
+    -   id: codeql-format
+        name: Fix QL file formatting
+        files: \.qll?$
+        language: system
+        entry: codeql query format --in-place
+    -   id: sync-files
+        name: Fix files required to be identical
+        language: system
+        entry: python3 config/sync-files.py --latest
```

Comment:
I'm a bit reluctant to suggest scripts that automatically rewrites your commits (I personally run my pre-commit hook for `codeql query format` without `--in-place`), but if people like automatic fixes like this I won't stand in their way 😄.

Reply:
Tomorrow I will have a look at the `pre-commit install --config` option, because that could provide an easy way to give the two flavours of hooks (auto-fixing vs only checking) already pre-made :slightly_smiling_face:

Reply:
yes, `--config` is definitely the way to go, I've updated `pre-commit-hook-setup.md` instructions accordingly.